org-roam.el

```diff
@@ -609,9 +669,21 @@ Applies `org-roam-link-face' if PATH correponds to a Roam file."
 (defun org-roam--find-file-hook-function ()
   "Called by `find-file-hook' when `org-roam-mode' is on."
   (when (org-roam--org-roam-file-p)
-    (org-link-set-parameters "file" :face 'org-roam--roam-link-face)
-    (add-hook 'post-command-hook #'org-roam--maybe-update-buffer nil t)
-    (add-hook 'after-save-hook #'org-roam--update-cache nil t)))
+    (setq org-roam-last-window (get-buffer-window))
+    (if (org-roam-cache-initialized)
+        (org-roam--setup-found-file)
+      (org-roam--build-cache-async
```

Comment:
what happens when you switch to multiple files in the same org-roam-directory with the cache still unbuilt? Will some of them be missing the hooks?

I think it's okay to setup the hooks first. `org-roam--update-cache` and `org-roam--maybe-update-buffer` can be a no-op when the caches aren't yet initialized.

Reply:
Unfortunately in testing it where the hooks are added first, I get this message if I try to add the hooks before the cache is built:
```
Error in post-command-hook (org-roam--maybe-update-buffer): (user-error "Your Org-Roam cache isn’t built yet! Please wait")
```

However, if I add a check for initialization inside of `org-roam--maybe-update-buffer`, this fixes the problem and I can move the hooks.  This also likely fixes the issue I've experience in testing where a buffer that has the hooks and the cache doesn't get built causes emacs to not be able to change buffers or exit.

Reply:
Yep, I think `org-roam--maybe-update-buffer` can also be a no-op if the caches aren't initialized.